src/lib/persistor.ts

```diff
@@ -1,13 +1,26 @@
+import { MoodleApi } from '../moodleapi'
+import { getCurrentContext } from './utils'
+
 export interface Persistor {
   get: (key: string) => Promise<string | null>
   put: (key: string, value: string) => Promise<void>
 }
 
-export const browserPersistor: Persistor = {
-  get: async (key: string): Promise<string | null> => {
-    return localStorage.getItem(key)
-  },
-  put: async (key: string, value: string) => {
-    localStorage.setItem(key, value)
+export const getMoodlePersistor = (): Persistor | undefined => {
+  const { courseId } = getCurrentContext()
```

Comment:
I know this is something we'd need to reflect in `main` later (e.g. to modify `events.ts` to use the data from context), but just to test / set the example here even though it's a spike branch, I'm wondering if it isn't better to just modify `getCurrentContext` so accesses to any of the magical moodle `window.M` variables is done there. Something like:

```javascript
export const getCurrentContext = (): { courseId: number | undefined, host: string | undefined, moodleWWWRoot: string | undefined, moodleSessKey: string | undefined } => {
  const courseId = window.M?.cfg.courseId
  const host = window.location.host
  const moodleWWWRoot = window.M?.cfg.wwwroot
  const moodleSessKey = window.M?.cfg.sesskey

  return {
    courseId,
    host,
    moodleWWWRoot,
    moodleSessKey
  }
```

I think doing this will help us encapsulate things a bit in terms of isolating where assumptions on "being in Moodle" get made.